packages/plugins/plugin-sheet/src/components/Sheet/util.ts

```diff
@@ -54,3 +63,34 @@ export const scrollIntoView = (scrollContainer: HTMLElement, el: HTMLElement) =>
     scrollContainer.scrollLeft += 2 + scrollerBounds.right - cellBounds.right;
   }
 };
+
+export type UseSheetModelProps = {
+  sheet: SheetType;
+  space: Space;
+  options?: Partial<FunctionContextOptions>;
+  readonly?: boolean;
+};
+
+export const useSheetModel = ({ space, sheet, options, readonly }: UseSheetModelProps) => {
+  const graph = useComputeGraph(space, options);
+
+  const [[model, formatting] = [], setModels] = useState<[SheetModel, FormattingModel] | undefined>(undefined);
+
+  useEffect(() => {
+    let model: SheetModel | undefined;
+    let formatting;
+    const t = setTimeout(async () => {
+      model = new SheetModel(graph, sheet, space, { readonly, mapFormulaBindingToId, mapFormulaBindingFromId });
+      await model.initialize();
+      formatting = new FormattingModel(model);
```

Comment:
inline declaration `const formatting = ` unless required at end to destroy?

Reply:
We can probably just set it anonymously.